### ISSUE TITLE
Conversion from qmd (specifying ir kernel) to ipynb create only markdown cells

### ISSUE THREAD

**Issue by statquant** [quarto-dev/quarto-cli#97]
Hello, I was trying `quarto` on the following example:

````
---                         
title: document_01          
author: statquant           
toc: true                   
toc-depth: 2                
format:                     
  html:                     
    max-width: 800px        
    fontsize: 18px          
    html-math-method: katex 
jupyter: ir                 
---                         
                            
That's a test               
                            
```{r}                      
#| echo: false              
plot(1:5)                   
```
````    

and running `quarto convert document_01.qmd` `document_01.ipynb` got generated but opened in `jupyterlab` as follows:

![image](https://user-images.githubusercontent.com/3612473/131367237-34d38f1d-acdb-48e9-adea-f3ea793640fa.png)

Is there additional yaml I should be adding ?

PS: 
- Sorry for the missing ticks in `document_01.qmd` I could not have them displayed properly
- The document mentions `quarto convert document.qmd --to xxx` while on the version I got it should say `quarto convert document.qmd --output xxx` (I think as the first errors while the second passes). Happy to PR this documentation change if that's any useful

**Comment by jjallaire** [comment]
Hi @statquant, thanks for reporting this!

It looks like one problem is that we are assuming that the Jupyter kernelspec `langauge` will be lower-case (in the case of R though it's "R", so our matching fails). That should be pretty easy to remedy on our end. I put in a quick fix for this and there was yet another issue which I haven't run down yet -- will ping you back here when I sort it out.

BTW you can include backtick code blocks here by encolosing them in four backticks like this:

`````
````
```{r}
#| echo: false              
plot(1:5)    
```
````
`````

Thx for the heads up on the `--to` argument, I'll make that change.

**Comment by jjallaire** [comment]
This should now be fixed on `main` and in the v0.2.109 binary release.

To fully support the `ir` kernel ideally we'd have setup and cleanup hooks implemented. Here they are for python:

https://github.com/quarto-dev/quarto-cli/tree/main/src/resources/jupyter/lang/python

For the setup file this amounts to:

1) Forwarding the default figure options (device, width, height, dpi) to the default graphics device and to htmlwidget contexts
2) Printing a hashable to cell output which when values change causes us to restart the kernel (we keep kernels alive across renders to mitigate the 2-3 second kernel startup time). For Python this is just a map of package compilation output file to last modified time -- this is so if a package is updated while the kernel is running we restart it.
3) Provide an implementation of the `ojs_define()` function (for interop w// observable: https://quarto.org/docs/interactive/ojs/#data-sources).

For the cleanup file this should just clear out state between runs (unload packages, clean the environment, reset options to defaults, etc.)

I don't know enough about the `ir` kernel to implement all of this, but we'd welcome a PR (note that the name of the folder should be `R` (capitalized) to match the `ir` kernel language name).

Note that things will still more or less work w/o all of the above, but it's way better if they are implemented.

**Comment by statquant** [comment]
Thank you for you prompt response.
I'll have a look (I have to confess most of your points sounds a bit like chinese to me for now).
